src/dgpost/utils/plot.py

```diff
@@ -0,0 +1,150 @@
+            y:      !!str       # column label for y data
+            x:      !!str       # optional column label for x data, or else index is used
+            kind:   !!str       # optional kind of plot to produce, default `scatter'
+                                # additional kwargs gets passed to plotting method
+          methods:  !!dict      # optional kwargs where the key is an attribute of an :class:`matplotlib.axes.Axes`
+                                # and the value is a dict containing kwargs to be called on the selected attribute
+                                # also possible to call method on subobject of the `axes` i.e. `axes.xaxis.set_label_text`
+        nrows:      !!int       # optional int defining the number of rows of the grid for :class:`matplotlib.gridspec.GridSpec`
+        ncols:      !!int       # optional int defining the number of columns of the grid for :class:`matplotlib.gridspec.GridSpec`
+        save:                   # optional to save the generated plot, `fname` required and any additional kwargs gets passed to :func:`matplotlib.figure.Figure.savefig`
+          as:       !!str       # path, where the file is saved
+          tight_layout: !!dict  # optional dictionary for figure.tight_figure
+
+.. codeauthor:: Ueli Sauter
+"""
+import matplotlib
+import pandas as pd
+import uncertainties.unumpy as unp
+from matplotlib import pyplot as plt
+from matplotlib.gridspec import GridSpec
+
+
+def apply_plot_style(style: dict):
+    """
+    Updates the plot style with the given dictionary.
+    For available kwargs see :class:`matplotlib.rcParams`.
+    If style is None, applies/resets to the default matplotlib style.
+
+    Parameters
+    ----------
+    style
+        A dictionary object containing valid key/value pairs.
+
+    """
+    if style is not None:
+        matplotlib.rcParams.update(style)
+    else:
+        matplotlib.rcdefaults()
+
+
+def plt_axes(ax: matplotlib.axes.Axes, table: pd.DataFrame, ax_args: dict):
+    """
+    Processes ax_args and plots the data
+
+    Parameters
+    ----------
+    ax
+        axes to be plotted to
+    table
+        dataframe containing the data
+    ax_args
+        arguments for the axes
+
+    """
+    series = ax_args.pop("series", [])
+
+    for method, arguments in ax_args.pop("methods", {}).items():
+        attr = ax
+        # allows for method calls on attributes
+        for met in method.split("."):
+            attr = getattr(attr, met)
+        attr(**arguments)
+    ax.set(**ax_args)
+
+    for element in series:
+        # datas processing
+        # should I use here separate_data and pQ from dgpost.transform.helpers?
+        # but with that I will lose the access to the index for x_data
+        y = element.pop("y")
+        y_data = table[y]
+        y_values = unp.nominal_values(y_data.array)
+        y_err = unp.std_devs(y_data.array)
+        y_unit = table.attrs.get("units", {}).get(y, "")
```

Comment:
About the unit: I think our original thoughts were that the `x` and `y label` automatically gets chosen based on the selected columns. But the `xlabel/ylabel` is an `axes` property and not a "line" property. We can also have multiple columns plotted inside a single `axes`. So I don't know if the `unit` is even necessary and the `x/ylabel` should be probably chosen manually.